lib/collection/src/collection/query.rs

```diff
@@ -60,6 +61,83 @@ impl Collection {
         Ok(results.into_iter().next().unwrap())
     }
 
+    /// If the query limit above this value, it will be a subject to undersampling.
+    const SHARD_QUERY_SUBSAMPLING_LIMIT: usize = 128;
```

Comment:
Why specifically this number? For segments we use `params.hnsw_ef`

Reply:
`hnsw_ef` was right choise for segment, cause we wanted to minimize HNSW overhead for large limits. This is not relevant for shard-level, where bottleneck is the networking